### PR TITLE
Adds a job to run istio-csr ECC e2e suite

### PR DESCRIPTION
**Pull request by SpectralHiss**
This runs the test-ecc e2e suite introduced in https://github.com/cert-manager/istio-csr/pull/274

### CODE REVIEW

**Review by SgtCoDFish — APPROVED** [review]
/lgtm
/approve
/hold

Adding a hold because once this merges it'll block all istio-csr PRs until the new test is merged - so we want to merge https://github.com/cert-manager/istio-csr/pull/274 ASAP after this merges.

**Comment by inteon** [comment]
https://github.com/cert-manager/istio-csr/pull/274 is (almost) ready
/unhold

**Pull request merged**
